wd_core.au3

```diff
@@ -1557,46 +1575,59 @@ EndFunc   ;==>__WD_Delete
 ; #INTERNAL_USE_ONLY# ===========================================================================================================
 ; Name ..........: __WD_Error
 ; Description ...: Writes Error to the console and show message-boxes if the script is compiled.
-; Syntax ........: __WD_Error($sWhere, $i_WD_ERROR[, $sMessage = Default])
+; Syntax ........: __WD_Error($sWhere, $iErr[, $sMessage = Default[, $iExt = Default]])
 ; Parameters ....: $sWhere     - Name of calling routine
-;                  $i_WD_ERROR - Error constant
-;                  $sMessage   - [optional] Additional Information (default = "")
-; Return values..: Success - Error Const from $i_WD_ERROR
+;                  $iErr       - The error number from the calling function
+;                  $sMessage   - Message that will be passed to the console/output [optional]
+;                  $iExt       - Extended information from the calling function [optional]
+; Return values..: Success - $iErr
 ;                  Failure - None
-; Author ........: Thorsten Willert, Dan Pollak
-; Modified ......:
+; Author ........: Stilgar, Danp2
+; Modified ......: mLipok
 ; Remarks .......:
 ; Related .......:
 ; Link ..........:
 ; Example .......: No
 ; ===============================================================================================================================
-Func __WD_Error($sWhere, $i_WD_ERROR, $sMessage = Default)
+Func __WD_Error($sWhere, $iErr, $sMessage = Default, $iExt = Default)
+	Local Const $sFuncName = "__WD_Error"
 	Local $sMsg
 
 	If $sMessage = Default Then $sMessage = ''
+	If $iExt = Default Then $iExt = 0
 
 	Switch $_WD_DEBUG
 		Case $_WD_DEBUG_None
 
 		Case $_WD_DEBUG_Error
-			If $i_WD_ERROR <> $_WD_ERROR_Success Then ContinueCase
+			If $iErr <> $_WD_ERROR_Success Then ContinueCase
 
 		Case $_WD_DEBUG_Info
-			$sMsg = $sWhere & " ==> " & $aWD_ERROR_DESC[$i_WD_ERROR]
-
-			If $sMessage <> "" Then
-				$sMsg = $sMsg & ": " & $sMessage
-			EndIf
-
+			Local $sExtended = (($iExt) ? (" / " & $iExt) : (""))
+			$sMsg = $sWhere & " ==> " & $aWD_ERROR_DESC[$iErr] & " ( " & $iErr & $sExtended & " )"
+			$sMsg &= (($sMessage) ? (" : " & $sMessage) : (""))
 			__WD_ConsoleWrite($sMsg)
 
-			If @Compiled Then
-				If $_WD_ERROR_MSGBOX And $i_WD_ERROR <> $_WD_ERROR_Success And $i_WD_ERROR < 6 Then MsgBox(16, "WD_Core.au3 Error:", $sMsg)
-				DllCall("kernel32.dll", "none", "OutputDebugString", "str", $sMsg)
+			If $iErr <> $_WD_ERROR_Success Then
+				If $_WD_ERROR_MSGBOX And $iErr < 6 Then

```

Comment:
I think we need to review this. Instead of "6", we could use the defined constant. Also, does this logic still apply? If so, then is 6 still the correct cutoff or has it changed as the UDF has grown?

Reply:
I go to bed.
I will answer tomorrow.

Reply:
I agree that this is things to review but not in this PR
So I create:
https://github.com/Danp2/au3WebDriver/issues/246

Could you take care of this problem this time?